src/components/common/Tour.js

```diff
@@ -0,0 +1,263 @@
+        content: PropTypes.string.isRequired,
+        disableBeacon: PropTypes.bool,
+      })
+    ).isRequired,
+    signInActivity: PropTypes.bool.isRequired,
+    visitSpaceTourRan: PropTypes.bool.isRequired,
+    settingsTourRan: PropTypes.bool.isRequired,
+    authenticated: PropTypes.bool.isRequired,
+    location: PropTypes.shape({
+      pathname: PropTypes.string,
+    }).isRequired,
+    history: PropTypes.shape({
+      replace: PropTypes.func.isRequired,
+      push: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  componentDidUpdate(prevProps) {
+    const {
+      signInActivity,
+      authenticated,
+      visitSpaceTourRan,
+      settingsTourRan,
+      dispatchResetTour,
+      dispatchInitializeTour,
+      location: { pathname },
+      currentTour,
+      run,
+    } = this.props;
+    const { signInActivity: prevSignInActivity } = prevProps;
+
+    if (
+      pathname === HOME_PATH &&
+      prevSignInActivity &&
+      !signInActivity &&
+      authenticated &&
+      !visitSpaceTourRan
+    ) {
+      setTimeout(
+        () =>
+          dispatchInitializeTour({
+            tour: tours.VISIT_SPACE_TOUR,
+            steps: VISIT_SPACE_TOUR_STEPS,
+          }),
+        TOUR_DELAY_750
+      );
+    }
+
+    if (prevSignInActivity && !signInActivity && !authenticated) {
+      dispatchResetTour();
+    }
+
+    if (
+      pathname === SETTINGS_PATH &&
+      authenticated &&
+      !settingsTourRan &&
+      currentTour !== tours.SETTINGS_TOUR &&
+      !run
+    ) {
+      setTimeout(
+        () =>
+          dispatchInitializeTour({
+            tour: tours.SETTINGS_TOUR,
+            steps: SETTINGS_TOUR_STEPS,
+          }),
+        TOUR_DELAY_750
+      );
+    }
+  }
+
+  render() {
+    const {
+      history: { push, replace },
+      tourKey,
+      run,
+      continuous,
+      stepIndex,
+      steps,
+      dispatchStartTour,
+      dispatchNextStep,
+      dispatchNavigateAndStopTour,
+      dispatchTourCompleted,
+      currentTour,
+    } = this.props;
+    const callback = async data => {
+      const { action, index, type, status } = data;
+      if (
+        action === ACTIONS.CLOSE ||
+        (status === STATUS.SKIPPED && run) ||
+        status === STATUS.FINISHED
+      ) {
+        dispatchTourCompleted(currentTour);
+      } else if (
+        type === EVENTS.STEP_AFTER ||
+        type === EVENTS.TARGET_NOT_FOUND
+      ) {
+        const newStepIndex = index + (action === ACTIONS.PREV ? -1 : 1);
+        switch (currentTour) {
+          case tours.VISIT_SPACE_TOUR:
+            switch (newStepIndex) {
+              case 1:
+                push(VISIT_PATH);
+                dispatchNavigateAndStopTour(newStepIndex);
+                await waitForElement({
+                  selector: `.${VISIT_SPACE_INPUT_CLASS}`,
+                });
+                document.getElementById(DRAWER_BUTTON_ID).click();
+                setTimeout(() => dispatchStartTour(), TOUR_DELAY_500);
```

Comment:
In this specific use-case, I used the `setTimeout` to wait for the sidebar to open completely. Otherwise, the tour box is looking distorted. I did indeed hide it under `waitForElement` though for other cases.

Reply:
Still, I would create a function in this class called `handleStartTour` and put this line inside.